:man_farmer: `launch_testing_ros` regressions `wait_for_topic_launch_test` on linux

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu `jammy`
- Installation type:
  - Source, buildfarm
- Version or commit hash:
  - Rolling, master
- DDS implementation:
  - Default: Fast-RTPS
- Client library (if applicable):
  - Probably `rclpy`

#### Steps to reproduce issue
Run a buildfarm job on `nightly_linux_debug`

#### Expected behavior
All tests pass.

#### Actual behavior
Various `launch_testing_ros` tests fail and throw exceptions.
See: https://ci.ros2.org/view/nightly/job/nightly_linux_debug/2251/

#### Additional information
`wait_for_topic_launch_test.TestFixture.test_topics_unsuccessful` shows:

<details><summary>Error log</summary>

```
Traceback (most recent call last):^M
  File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/wait_for_topic_launch_test.py", line 89, in test_topics_unsuccessful^M
    assert wait_for_node_object.topics_received() == expected_topics^M
AssertionError: assert {'chatter_0',..., 'chatter_4'} == {'chatter_0',..., 'chatter_4'}^M
  Extra items in the right set:^M
  'chatter_2'^M
  'chatter_1'^M
  Use -v to get more diff

...

[INFO] [python-7]: sending signal 'SIGINT' to process[python-7]^M
[INFO] [python-6]: sending signal 'SIGINT' to process[python-6]^M
[python-6] Traceback (most recent call last):^M
[python-6]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/talker.py", line 51, in <module>^M
[python-6]     main()^M
[python-6]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/talker.py", line 47, in main^M
[python-6]     rclpy.shutdown()^M
[python-6]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/__init__.py", line 126, in shutdown^M
[python-6]     _shutdown(context=context)^M
[python-6]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/utilities.py", line 58, in shutdown^M
[python-6]     return context.shutdown()^M
[python-6]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/context.py", line 102, in shutdown^M
[python-6]     self.__context.shutdown()^M
[python-6] rclpy._rclpy_pybind11.RCLError: failed to shutdown: rcl_shutdown already called on the given context, at /home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/rcl/rcl/src/rcl/ini\
t.c:241

```
</details>

`wait_for_topic_launch_test.TestFixture.test_topics_successful` shows:

<details> <summary>Error log</summary>

```
Traceback (most recent call last):^M
  File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/wait_for_topic_launch_test.py", line 61, in test_topics_successful^M
    with WaitForTopics(topic_list, timeout=2.0) as wait_for_node_object_1:^M
  File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/launch_testing_ros/wait_for_topics.py", line 96, in __enter__^M
    raise RuntimeError('Did not receive messages on these topics: ',^M
RuntimeError: ('Did not receive messages on these topics: ', {'chatter_4'})
...
[python-12] Traceback (most recent call last):^M
[python-12]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/talker.py", line 51, in <module>^M
[python-12]     main()^M
[python-12]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/talker.py", line 47, in main^M
[python-12]     rclpy.shutdown()^M
[python-12]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/__init__.py", line 126, in shutdown^M
[python-12]     _shutdown(context=context)^M
[python-12]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/utilities.py", line 58, in shutdown^M
[python-12]     return context.shutdown()^M
[python-12]   File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/install/rclpy/lib/python3.10/site-packages/rclpy/context.py", line 102, in shutdown^M
[python-12]     self.__context.shutdown()^M
[python-12] rclpy._rclpy_pybind11.RCLError: failed to shutdown: rcl_shutdown already called on the given context, at /home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/rcl/rcl/src/rcl/in\
it.c:241
...
Traceback (most recent call last):^M
  File "/home/jenkins-agent/workspace/nightly_linux_debug/ws/src/ros2/launch_ros/launch_testing_ros/test/examples/wait_for_topic_launch_test.py", line 69, in test_topics_successful^M
    assert wait_for_node_object_2.wait()^M
AssertionError: assert False^M
 +  where False = <bound method WaitForTopics.wait of <launch_testing_ros.wait_for_topics.WaitForTopics object at 0x7f02e01ab190>>()^M
 +    where <bound method WaitForTopics.wait of <launch_testing_ros.wait_for_topics.WaitForTopics object at 0x7f02e01ab190>> = <launch_testing_ros.wait_for_topics.WaitForTopics object at 0x7f02e01a\
b190>.wait^M
^M
```
</details>

I'm able to reproduce both failures ( `assert wait_for_node_object_2.wait()` and `AssertionError: assert set() == {'chatter_0',..., 'chatter_4'}` , but only when running `stress --cpu N`. With an idle system I can't reproduce at all.

--- a/launch_testing_ros/test/examples/talker.py
+++ b/launch_testing_ros/test/examples/talker.py
@@ -23,8 +23,9 @@ class Talker(Node):
     def __init__(self):
         super().__init__('talker')
         self.count = 0
+        hz_param = self.declare_parameter('hz', 1.0)
         self.publisher = self.create_publisher(String, 'chatter', 10)
-        self.timer = self.create_timer(1.0, self.callback)
+        self.timer = self.create_timer(1.0 / hz_param.value, self.callback)
 
     def callback(self):
         data = 'Hello World: {0}'.format(self.count)

--- a/launch_testing_ros/test/examples/wait_for_topic_launch_test.py
+++ b/launch_testing_ros/test/examples/wait_for_topic_launch_test.py
@@ -34,6 +34,7 @@ def generate_node(i):
         arguments=[os.path.join(path_to_test, 'talker.py')],
         name='demo_node_' + str(i),
         additional_env={'PYTHONUNBUFFERED': '1'},
+        parameters=[{'hz': 10.0}],
         remappings=[('chatter', 'chatter_' + str(i))]
     )
 

